types/react/index.d.ts

```diff
@@ -136,8 +136,12 @@ declare namespace React {
         type: ClassType<P, T, C>): CFactory<P, T>;
     function createFactory<P>(type: ComponentClass<P>): Factory<P>;
 
-    function createElement<P extends DOMAttributes<T>, T extends Element>(
-        type: string,
+    function createElement<P extends HTMLAttributes<T>, T extends Element>(
+        type: keyof ReactHTML,
+        props?: ClassAttributes<T> & P,
+        ...children: ReactNode[]): DOMElement<P, T>;
```

Comment:
Yes, but only if `T` is restricted to extend `HTMLElement`. I think this is fine to change, though.